modules/related-posts/jetpack-related-posts.php

```diff
@@ -270,82 +357,45 @@ public function render_block( $attributes ) {
-							'jp-relatedposts-post',
-							'jp-relatedposts-post' . $index,
-							! empty( $block_attributes['show_thumbnails'] ) ? 'jp-relatedposts-post-thumbs' : '',
-						)
-					);
-					$title_attr = $related_post['title'];
-					if ( '' !== $related_post['excerpt'] ) {
-						$title_attr .= "\n\n" . $related_post['excerpt'];
-					}
-					?>
-					<div
-						class="<?php echo esc_attr( implode( ' ', $classes ) ); ?>"
-						data-post-id="<?php echo esc_attr( $related_post['id'] ); ?>"
-						data-post-format="<?php echo esc_attr( ! empty( $related_post['format'] ) ? $related_post['format'] : 'false' ); ?>"
-					>
-						<?php if ( ! empty( $block_attributes['show_thumbnails'] ) && ! empty( $related_post['img']['src'] ) ) : ?>
-							<a class="jp-relatedposts-post-a"
-								href="<?php echo esc_url( $related_post['url'] ); ?>"
-								title="<?php echo esc_attr( $title_attr ); ?>"
-								rel="<?php echo esc_attr( $related_post['rel'] ); ?>"
-								data-origin="<?php echo esc_attr( $related_post['url_meta']['origin'] ); ?>"
-								data-position="<?php echo esc_attr( $related_post['url_meta']['position'] ); ?>"
-							>
-								<img class="jp-relatedposts-post-img"
-									src="<?php echo esc_url( $related_post['img']['src'] ); ?>"
-									width="<?php echo esc_attr( $related_post['img']['width'] ); ?>"
-									alt="<?php echo esc_attr( $title_attr ); ?>"
-								/>
-							</a>
-						<?php endif; ?>
-
-						<h4 class="jp-relatedposts-post-title">
-							<a
-								class="jp-relatedposts-post-a"
-								href="<?php echo esc_url( $related_post['url'] ); ?>"
-								title="<?php echo esc_attr( $title_attr ); ?>"
-								rel="<?php echo esc_attr( $related_post['rel'] ); ?>"
-								data-origin="<?php echo esc_attr( $related_post['url_meta']['origin'] ); ?>"
-								data-position="<?php echo esc_attr( $related_post['url_meta']['position'] ); ?>"
-							>
-								<?php echo esc_html( $related_post['title'] ); ?>
-							</a>
-						</h4>
-
-						<p class="jp-relatedposts-post-excerpt"><?php echo esc_html( $related_post['excerpt'] ); ?></p>
-
-						<?php if ( $block_attributes['show_date'] ) : ?>
-							<p class="jp-relatedposts-post-date" style="display: block;">
-								<?php echo esc_html( $related_post['date'] ); ?>
-							</p>
-						<?php endif; ?>
-
-						<?php if ( $block_attributes['show_context'] ) : ?>
-							<p class="jp-relatedposts-post-context">
-								<?php echo esc_html( $related_post['context'] ); ?>
-							</p>
-						<?php endif; ?>
-					</div>
-				<?php endforeach; ?>
-			</div>
-		</div>
+		<nav
+				class="jp-relatedposts-i2"
+				data-layout="<?php echo esc_attr( $block_attributes['layout'] ); ?>"
+		>
+			<?php
+			$this->render_block_row( $upper_row_posts, $block_attributes );
+			if ( $display_lower_row ) {
+				$this->render_block_row( $lower_row_posts, $block_attributes );
+			}
+			?>
+		</nav>
 		<?php
 		$html = ob_get_clean();
 
+		remove_filter( 'the_content', 'wpautop' );
```

Comment:
I unsure about this change.
Could we achieve the same result by hiding `<p>`s via css.
or as @enejb suggested maybe replacing new lines from `$html` after `$html = ob_get_clean();`?

Reply:
I can modify this a bit — or simply remove it in anticipation for https://core.trac.wordpress.org/ticket/45495 being closed in time.

Reply:
Could you provide a bit of background on this change? Is this already an issue with the current version of the Related Posts block? If yes, how can I reproduce the problem? If not, what changed?

Reply:
Internal discussion: pafL3P-jr-p2